File: backend/src/main/java/com/smart_parking_system/backend/entity/ParkingSpace.java
package com.smart_parking_system.backend.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "parking_space")
public class ParkingSpace {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ps_id", nullable = false)
    private Integer id;

    @Column(name = "name", length = Integer.MAX_VALUE)
    private String name;

    @Column(name = "location", length = Integer.MAX_VALUE)
    private String location;

    @Column(name = "owner", nullable = false, length = Integer.MAX_VALUE)
    private String owner;
}

File: mobile/app/src/main/java/com/smartparking/mobile/data/model/ParkingSpace.kt
package com.smartparking.mobile.data.model

data class ParkingSpace(
    val id: Long,
    val name: String,
    val location: String,
    val owner: String? = null
)

data class CreateParkingSpaceRequest(
    val name: String,
    val location: String
)

data class UpdateParkingSpaceRequest(
    val name: String,
    val location: String
)
